src/main/java/com/guciowons/shoppingbasket/Basket/BasketService.java

```diff
@@ -0,0 +1,45 @@
+package com.guciowons.shoppingbasket.Basket;
+
+import com.guciowons.shoppingbasket.Product.ProductDao;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BasketService {
+    private Basket basket = new Basket();
+
+    private final ProductDao productDao;
+    private final BasketSummarizer basketSummarizer;
+
+    public BasketService(ProductDao productDao, BasketSummarizer basketSummarizer) {
+        this.productDao = productDao;
+        this.basketSummarizer = basketSummarizer;
+    }
+
+    public ResponseEntity<String> addProductToBasket(int id, int quantity) {
+        return productDao.findById(id)
+                .map(product -> {
+                    basket.addProduct(id, quantity);
+                    return new ResponseEntity<>("Done", HttpStatus.ACCEPTED);
+                }).orElseGet(() -> new ResponseEntity<>("No such product", HttpStatus.NOT_FOUND));
```

Comment:
Cała struktura o wiele lepiej, ale mieszasz tutaj dwa obszary. Warstwa serwisu nie powinna wiedzieć nic o tym, że istnieje coś takiego jak warstwa RESTa, wszystko powinno być niezależne. Serwis powien dostać dane, coś z nimi zrobić i zwrócić wynik, albo rzucić wyjątek. Tutaj zwracasz obiekty `ResponseEntity`, które należą właśnie do warsty RESTa, przez co ten serwis jest uzależniony od tej technologii komunikacji.

Przy poprawnym dodaniu nie zwracaj nic. Przy błędzie rzuć tutaj np jakiś `IllegalArgumentException` i wtedy obsłuż go w `BasketController`, przez rzucenie właśnie `not found`.